### PR TITLE
UTOPIA-1308: (backend) add new status

### PR DESCRIPTION
**Pull request by Bottle7**
# Description

This PR includes the following proposed change(s):

- CPO can now move a PIA from 'Pending completion' => 'Complete'

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Move a new Delegated PIA from 'Incomplete' => 'Complete' 

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by binggaofw** [comment]
do you want to add unit test in a different PR or will add the ut later?

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 1]
```diff
@@ -283,8 +283,63 @@ export const piaStatusMetadata: Partial<
...
+  [PiaIntakeStatusEnum.PENDING_COMPLETION]: {
+    updates: {
+      allow: false,
+    },
+    transition: {
+      [PiaIntakeStatusEnum.INCOMPLETE]: {
+        allow: true,
+        conditions: [
+          {
+            accessType: [UserTypesEnum.CPO],
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:300]: Is pending completion applicable to Delegate PIA? If not, can we add piaType=STANDARD condition as well?

<applicable to all transitions>

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 2]
```diff
@@ -283,8 +283,63 @@ export const piaStatusMetadata: Partial<
...
+          },
+        ],
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:302]: On transitioning from PENDING_COMPLETION to Edit statuses, review needs to be deleted.

```suggestion
        ],
        actions: [
          {
            type: 'update',
            key: 'review',
            value: null,
          },
        ],
```

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 3]
```diff
@@ -283,8 +283,63 @@ export const piaStatusMetadata: Partial<
...
+      },
+      [PiaIntakeStatusEnum.EDIT_IN_PROGRESS]: {
+        allow: true,
+        conditions: [
+          {
+            accessType: [UserTypesEnum.CPO],
+          },
+        ],
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:310]:
```suggestion
        ],
        actions: [
          {
            type: 'update',
            key: 'review',
            value: null,
          },
        ],
```

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 4]
```diff
@@ -283,8 +283,63 @@ export const piaStatusMetadata: Partial<
...
         ],
       },
+      [PiaIntakeStatusEnum.PENDING_COMPLETION]: {
+        allow: true,
+      },
+    },
+  },
+  [PiaIntakeStatusEnum.PENDING_COMPLETION]: {
+    updates: {
+      allow: false,
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:293]: Missing override to allow commenting. Correct me if I am wrong, but I think comments are allowed in Pending Completion.

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 5]
```diff
@@ -283,8 +283,63 @@ export const piaStatusMetadata: Partial<
           },
         ],
       },
+      [PiaIntakeStatusEnum.PENDING_COMPLETION]: {
+        allow: true,
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:287]: <optional> Do you think if we should add field validations to check if all Program Area reviews are submitted? And only then you're allowed to move to PENDING_COMPLETION status

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts (hunk 1 above)
Comment (Bottle7) [reply src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:300]: I'm pretty sure that, in either case, Pending Completion is a valid status before Complete. In a Delegated flow, a CPO would still have to upload the PIA to the PID. Until they do so, the PIA would be 'Pending completion."

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts (hunk 4 above)
Comment (Bottle7) [reply src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:293]: Comments are not allowed in the 'Pending completion' status, to my knowledge. The PIA is _effectively_ complete, it is simply pending upload to the PID by a CPO.

Comment (binggaofw) [reply src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:293]: yes, comments does not allow in pending and complete status, I can recall that

Comment (kushal-arora-fw) [reply src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:293]: Great, thanks 👍

**Review by BradyMitch — APPROVED** [review]

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 6]
```diff
@@ -283,8 +283,133 @@ export const piaStatusMetadata: Partial<
...
+            accessType: [UserTypesEnum.CPO],
+          },
+        ],
+      },
       [PiaIntakeStatusEnum.COMPLETE]: {
         allow: true,
+        conditions: [
+          {
+            accessType: [UserTypesEnum.CPO],
+          },
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:384]: I doubt this is true. Any CPO user should not be able to move from Pending Completion to COMPLETE without conditions. Suggested addition of User Types to below conditions. And recommend removing this one.

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 7]
```diff
@@ -283,8 +283,133 @@ export const piaStatusMetadata: Partial<
...
+                required: true,
+              },
+              'review.cpo': {
+                required: true,
+              },
+              'review.programArea.reviews': {
+                required: true,
+                size: SizeEnum.ALL,
+              },
+            },
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:410]:
```suggestion
            piaType: [PiaTypesEnum.STANDARD],
            accessType: [UserTypesEnum.CPO],
            fieldValidations: {
              'review.mpo.reviewNote': {
                required: true,
              },
              'review.cpo': {
                required: true,
              },
              'review.programArea.reviews': {
                required: true,
                size: SizeEnum.ALL,
              },
            },
```

File: src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts [hunk 8]
```diff
@@ -283,8 +283,133 @@ export const piaStatusMetadata: Partial<
...
+            fieldValidations: {
+              'review.mpo.reviewNote': {
+                required: true,
+              },
+              'review.programArea.reviews': {
+                required: true,
+                size: SizeEnum.ALL,
+              },
+            },
+          },
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/metadata/pia-status.metadata.ts:396]:
```suggestion
            piaType: [PiaTypesEnum.DELEGATE_REVIEW],
            accessType: [UserTypesEnum.MPO],
            fieldValidations: {
              'review.mpo.reviewNote': {
                required: true,
              },
              'review.programArea.reviews': {
                required: true,
                size: SizeEnum.ALL,
              },
            },
          },
```

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]

**Review by kushal-arora-fw — APPROVED** [review]
Fantastic-o!

**Review by binggaofw — APPROVED** [review]

**Pull request merged**